lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -0,0 +1,189 @@
+from lstchain.io import read_mc_dl2_to_pyirf, read_configuration_file, get_standard_config
+from lstchain.reco.utils import filter_events
+
+from astropy.io import fits
+import astropy.units as u
+
+from pyirf.io.gadf import create_aeff2d_hdu, create_energy_dispersion_hdu
+from pyirf.irf import effective_area_per_energy, energy_dispersion, effective_area_per_energy_and_fov
+from pyirf.utils import calculate_source_fov_offset, calculate_theta
+from pyirf.binning import create_bins_per_decade, add_overflow_bins
+
+__all__ = [
+    'IRFFITSWriter'
+]
+
+class IRFFITSWriter(Tool):
+    name = "IRFFITSWriter"
+    description = "Create IRF FITS file from given MC DL2 files and selection cuts"
+
+    input_gamma_dl2 = traits.Path(
+        help="Input MC gamma DL2 file", exists=True, directory_ok=False, file_ok=True
+    ).tag(config=True)
+    output_irf_path = traits.Path(
+        help="IRF output filedir",
+        directory_ok=True, file_ok=False
+    ).tag(config=True)
+    pnt_like = traits.Bool(
+        help="True for point-like IRF, False for Full Enclosure",
+        default_value=False,
+    ).tag(config=True)
+    config_file = traits.Path(
+        help="Config file for selection cuts", directory_ok=False, file_ok=True,
+    ).tag(config=True)
+    input_gamma_diff_dl2 = traits.Path(
+        help="Input MC diffuse gamma DL2 file", exists=True, directory_ok=False,
+        file_ok=True
+    ).tag(config=True)
+
+    aliases = {
+        "input_gamma_dl2" : "IRFFITSWriter.input_gamma_dl2",
+        "fg" : "IRFFITSWriter.input_gamma_dl2",
+        "output_irf_path" : "IRFFITSWriter.output_irf_path",
+        "o" : "IRFFITSWriter.output_irf_path",
+        "pnt_like" : "IRFFITSWriter.pnt_like",
+        "pnt" : "IRFFITSWriter.pnt_like",
+        "config_file" : "IRFFITSWriter.config_file",
+        "conf" : "IRFFITSWriter.config_file",
+        "input_gamma_diff_dl2" : "IRFFITSWriter.input_gamma_diff_dl2",
+        "fg_diff" : "IRFFITSWriter.input_gamma_diff_dl2",
+    }
+
+    flag = {
+        "pnt_like": (
+        {"IRFFITSWriter": {"pnt_like":False}},
+        "Full Enclosure IRFs will be produced"
+        ),
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that creates a compressed FITS file of IRFs for a given MC DL2 file
+        and with the given selection cuts.
+        For getting help run:
+        lstchain_create_irf_files --help
+        """
+        self.mc_gamma = None
+        self.cuts = None
+        self.hdus = None
+        self.effective_area = None
+        self.edisp = None
+        self.output_file = None
+
+    def setup(self):
+
+        if self.config_file is None:
+            self.cuts = read_configuration_file(os.path.join(os.path.dirname(__file__), '../data/data_selection_cuts.json'))
+        else:
+            self.cuts = read_configuration_file(self.config_file)
+
+        if str(self.input_gamma_dl2) is not None:
```

Comment:
Why is that string conversion necessary?

Reply:
This if is never true because str will never return `None`

Reply:
I incorrectly assumed that it was working. On this, I would like to ask if we should have a separate argument for diffuse gamma or not. One idea I have is to use a boolean as an argument, just like for point-like IRF marker, and to include these markers in the config file itself.
Kindly let me know if you have any suggestions for including it in this PR.

Reply:
You should be able to deduce from the simulation header (view_cone) what kind of file it is. You shouldn't need to specify it as a cli argument.

Reply:
Amended the files accordingly.